veps/sig-compute/secure-execution.md

```diff
@@ -0,0 +1,59 @@
+# Overview
+Implement support for IBM Z Secure Execution VMs in kubevirt
+
+## Motivation
+To enable customers to deploy secure workloads on their IBM Z machines using kubevirt.
+
+## Goals
+To enable kubevirt to run Secure Execution VMs on s390x
+
+## Non Goals
+Changing how existing VMs run.
+
+## Definition of Users
+- IBM Z users wanting to secure their new VMs using Secure Execution
+- IBM Z users wanting to migrate existing Secure Execution VMs to kubevirt
+
+## User Stories
+(list of user stories this design aims to solve)
+
+## Repos
+https://github.com/kubevirt/kubevirt
+https://github.com/kubevirt/api
+
+# Design
+For implementing this feature, the api would need to be expanded to have a Secure Execution option under Launch Security next to AMD SEV.
+Similiar to SEV it would be first behind a Feature Gate.
+
+Afterwards kubevirt would need to be expanded to handle setting the correct Launch Security type for Secure Execution.
+Lastly virt-controller would need to be modified to support starting virt-launcher pods with the correct capabilities and devices needed for Secure Execution.
+
+## API Examples
+```
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  ...
+spec:
+  domain:
+    launchSecurity:
+      secureExecution: {}
+  ...
+```
+
+## Scalability
+It should scale like normal VMs.
+The only restriction would be for Secure Execution VMs, as they ´can't run on IBM Z machines that they have not been encrypted with.
+To move a Secure Execution VM to a new IBM Z machine, it's public key would first need to be added to the keys used to encrypt the kernel+initramfs+parmfile.
+
+## Security
+The virt-launcher pod would currently need to be run as privileged for Secure Execution VMs. Libvirt checks for `/sys/firmware/uv` when verifying that the Host supports Secure Execution VMs. Unfortunately `/sys/firmware` is intentionally empty in unprivileged pods and kubernetes prevents mounting anything under this path. There are potentially Feature Gates in kubelet to allow this, however i have not yet been able to get this to work. From what i gathered you would need to at least enable `ProcMountType` and `UserNamespacesSupport` to be able to launch a Secure Execution VM inside of an unprivileged pod.
```

Comment:
This seems to be a huge no-go. I think we have dropped the root support for VMs

Reply:
We should not touch or require any unsecure kubernetes feature gate or option. You could try to implement a device plugging for secure execution which mounts those directory in the pod. If it isn't a DP then a similar mechanism

Reply:
It is still in there for ppc64le

Reply:
you can do the bindmounts from the privileged DP. Not very kubernetes friendly, but I'm pretty sure you can do it. 
For device pluging you can specify the hostpath and the container path. So, at least, I would like to have those 2 ways explored. 
As I said, running VMs as privileged makes this proposal not acceptable, so we need to find an alternative to it

Reply:
Device plugin works since we are using it for mounting a socket for example. See [here](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/device-manager/socket_device.go#L169). Bind mounts we use them for container and hotplug disks. The right tool to take a look at is [virt-chroot](https://github.com/kubevirt/kubevirt/tree/main/pkg/virt-handler/virt-chroot). You can see as example, the [container disks](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/container-disk/mount.go) and [hotplug disk](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/container-disk/mount.go) in the handler